packages/botonic-react/src/webchat/textarea/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from 'react'
+import TextareaAutosize from 'react-textarea-autosize'
+
+// import { PersistentMenuTheme } from '../../components/index-types'
+import { WEBCHAT } from '../../constants'
+import { WebchatContext } from '../../contexts'
+import { DeviceAdapter } from '../devices/device-adapter'
+import { TextAreaContainer } from './styles'
+
+interface TextareaProps {
+  deviceAdapter: DeviceAdapter
+  persistentMenu: any //PersistentMenuTheme
+  textareaRef: React.MutableRefObject<HTMLTextAreaElement>
+  sendChatEvent: (event: string) => Promise<void>
+  sendTextAreaText: (text: string) => Promise<void>
+}
+
+export const Textarea = ({
+  deviceAdapter,
+  persistentMenu,
+  textareaRef,
+  sendChatEvent,
+  sendTextAreaText,
+}: TextareaProps) => {
+  const { getThemeProperty } = useContext(WebchatContext)
+
+  let isTyping = false
+  let typingTimeout
```

Comment:
i know this was probably done like this. not for this PR, but I think we should use refs instead of direct variables